RocketChat/Rocket.Chat.ReactNative
# [FIX] Spotlight returning duplicated entries

## Proposed changes


The search method used to look for users in the RoomsListView requires to filter the users we already have a chat with. This is required, since this method looks for every user in the app, so when we're looking for a name, at search we will see the open chat with that person, the users with similar username and the username of the same person which we already have a chat open, and this is not the correct behavior for this.

## How to test or reproduce


In order to test this, you should be logged in a server.

- While being on the main screen, tap on the search button
- Type any username and open a chat with the one you looked for
- Go back to the main screen and search for the same username again

## Screenshots
<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</thead>
</tr>
<tbody>
<tr>
<td>
<img width="366" alt="image" src="https://user-images.githubusercontent.com/36211892/104762735-3a6f6d00-573b-11eb-8a46-3d59bfd9e121.png">
</td>

<td>
<img width="372" alt="image" src="https://user-images.githubusercontent.com/36211892/104766055-4f023400-5740-11eb-831e-d4b877dddf4f.png">
</td>
</tr>
</tbody>
</table>

## Types of changes




## Checklist
